schemas/dab.draft.schema.json

```diff
@@ -236,6 +236,22 @@
               }
             }
           }
+        },
+        "cache": {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "enabled": {
```

Comment:
with this new schema in place, will the `dab validate` command be successful if used on a config that contains the cache property?

Reply:
Working on validating this. Odd error that states schema references were not all resolved properly. Doesn't seem related to `cache` because cli still says "json validated" but an exception is thrown from the validate function which checks config against schema

Reply:
![image](https://github.com/Azure/data-api-builder/assets/6414189/c5dad3a5-3389-4469-b55b-6870edaa796f)

The JSON Schema validation tooling does not like that we have the following property and value in our `dab-config.json` file:
```json
  "$schema": "https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json",
```

Even when the link IS valid like `https://raw.githubusercontent.com/Azure/data-api-builder/37c6db7eb23edcd5b9398e4751951b79da26a73f/schemas/dab.draft.schema.json` the error is still encountered.

when i hard code DAB's json schema in code and exclude the `$schema` property and url value from the runtime config file, i get the expected behavior
```csharp
    public async Task<string?> GetJsonSchema(RuntimeConfig runtimeConfig)
    {
        // DEFAULT_CONFIG_SCHEMA_LINK is just a placeholder with no actual schema, hence should not be used to fetch the schema.
        if (!string.IsNullOrWhiteSpace(runtimeConfig.Schema) && !runtimeConfig.Schema.Equals(RuntimeConfig.DEFAULT_CONFIG_SCHEMA_LINK))
        {
            try
            {
                JsonSchema jsonSchema = await JsonSchema.FromUrlAsync("https://raw.githubusercontent.com/Azure/data-api-builder/37c6db7eb23edcd5b9398e4751951b79da26a73f/schemas/dab.draft.schema.json");
                return jsonSchema.ToJson();
            }
            catch (Exception e)
            {
                _logger!.LogError($"Failed to get schema from url: {runtimeConfig.Schema}\n{e}");
            }
        }
```

Success message: 
![image](https://github.com/Azure/data-api-builder/assets/6414189/6f3db295-92d3-4e80-a8aa-4664ffed4ccb)

Information: The config satisfies the schema requirements.

- Error: Failed to get schema from url: https://raw.githubusercontent.com/Azure/data-api-builder/37c6db7eb23edcd5b9398e4751951b79da26a73f/schemas/dab.draft.schema.json

> System.InvalidOperationException: **Could not find the JSON path of a referenced schema: Manually referenced schemas must be added to the 'Definitions' of a parent schema.**

>    at NJsonSchema.JsonPathUtilities.**GetJsonPaths**(Object rootObject, IEnumerable`1 searchedObjects, IContractResolver contractResolver)
> ...
>    at Azure.DataApiBuilder.Core.Configurations.JsonConfigSchemaValidator.GetJsonSchema(RuntimeConfig runtimeConfig) in **C:\Users\seleonar\Documents\dev\dabquickpr\src\Core\Configurations\JsonConfigSchemaValidator.cs:line 75**

Reply:
All that being said, we'll need to have a follow up deeper look into what if anything needs to be changed.

Reply:
when the schema fetching from url step fails, it gets the schema from the package itself, and it should do the validation giving the expected result. apart from the url fetch failure, did you see any difference wrt to using the hard-coded json schema?

Reply:
I have not been able to get URL schema fetching to succeed in any circumstances. 
However, pulling from assembly works. I'm not sure what's wrong with pulling from url. the ones I've used are accessible and not the unversioned sample link.

Reply:
#2013